src/rgw/rgw_rest_user_policy.cc

```diff
@@ -120,6 +120,7 @@ void RGWPutUserPolicy::execute(optional_yield y)
   bufferlist bl = bufferlist::static_from_string(policy);
 
   std::unique_ptr<rgw::sal::User> user = store->get_user(rgw_user(user_name));
+  uint64_t max_pol_limit = store->get_inline_policy_limit();
```

Comment:
Update the code to directly refer the policy_limit. See below code snippet :
  CephContext *cct = store->ctx();
  size_t max_pol_size = cct->_conf->rgw_user_policy_limit;